### PR TITLE
stylo: Use AtomicUsize to store our initial computed values

### PR DESCRIPTION
**Pull request by bholley**
This addresses the discussion in https://github.com/servo/servo/pull/12725/files/b916ad0a85b88eb40d919986cdc0c6b7420a4fab#r73459385

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12750)

### CODE REVIEW

File: components/style/properties/gecko.mako.rs [hunk 1]
```diff
@@ -1362,7 +1365,16 @@ ${impl_style_struct(style_struct)}
 ${define_ffi_struct_accessor(style_struct)}
 % endfor
 
-static mut INITIAL_GECKO_VALUES: *mut ComputedValues = 0 as *mut ComputedValues;
+// To avoid UB, we store the initial values as a atomic. It would be nice to
+// store them as AtomicPtr, but we can't have static AtomicPtr without const
+// fns, which aren't in stable Rust.
+static INITIAL_VALUES_STORAGE: AtomicUsize = ATOMIC_USIZE_INIT;
+unsafe fn raw_initial_values() -> *mut ComputedValues {
```
Comment (emilio) [inline components/style/properties/gecko.mako.rs]: I still haven't found any reference to undefined behavior for doing what we were doing, but anyway, I've verified that using `AtomicPtr` with `Ordering::Relaxed` is only 1 instruction more in x86.

@bors-servo: r+

**Comment by emilio** [comment]
Well, bors doesn't peek inline comments. Feel free to r=me, or waiting for @Manishearth comments if any :)

**Comment by Manishearth** [comment]
@bors-servo r+

I don't particularly like using atomicusize as a stand in for atomicptr, but AIUI they have the same semantics so it's fine.

**Pull request merged**